samples/tests/traits/mod.rs

```diff
@@ -0,0 +1,43 @@
+samples::test! {
+    volumetric;
+    /// ANCHOR: volumetric
+    trait Volumetric {
+        /// Strain energy density
+        fn psi(&self, j: f64) -> f64;
+        /// Derivative of strain energy with respect to $J$
+        fn d_psi(&self, j: f64, b_psi: f64) -> (f64, f64);
```

Comment:
I'd like for implementers of this trait to have a choice of implementing this function directly or via AD.

Reply:
Are default implementation a thing in Rust? I found this, but it looks not exactly like it? https://doc.rust-lang.org/book/ch10-02-traits.html#default-implementations
If not it would require independent t-lang approval since it would be a language change. I will leave it as is for now.